kube-runtime/src/watcher.rs

```diff
@@ -33,6 +33,8 @@ pub enum Error {
         source: kube::Error,
         backtrace: Backtrace,
     },
+    #[snafu(display("too many objects matched search criteria"))]
+    TooManyObjects { backtrace: Backtrace },
```

Comment:
This is only possible for `watch_object`, so maybe it should be broken out into a separate enum?

Reply:
So you'd like to move `watch_object` into `wait` as well? That doesn't feel quite right, since it's not really waiting-specific...

Reply:
Yeah, it might be the wrong thing to do. It feels a bit awkward inside watcher module as well though.

Maybe it's just the naming that's making it awkward. What do you think about restructuring names to something like:

`kube_runtime`
- `watch`
  * `collection` (aka `watcher`)
  * `single` (aka `watch_object`)


so that people can do `kube_runtime::watch::collection`?